pkg/ebpf/common/http/elasticsearch_test.go

```diff
@@ -0,0 +1,93 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package ebpfcommon
+
+import (
+	"net/http"
+	"net/http/httptest"
+	"strings"
+	"testing"
+)
+
+func TestParseElasticsearchRequest(t *testing.T) {
+	newRequest := func(method, target, body string) *http.Request {
+		return httptest.NewRequest(method, target, strings.NewReader(body))
+	}
+
+	tests := []struct {
```

Comment:
Can we add tests for various malformed bodies and incomplete queries? It can happen with partial buffers and customers sending wrong requests.